src/main/java/com/flightstats/hub/dao/aws/S3LargeContentDao.java

```diff
@@ -199,18 +199,20 @@ private Content getS3Object(String channelName, ContentKey key) throws IOExcepti
         long start = System.currentTimeMillis();
         try {
             GetObjectRequest request = new GetObjectRequest(bucketName, getS3ContentKey(channelName, key, false));
-            S3Object object = s3Client.getObject(request);
-            ObjectMetadata metadata = object.getObjectMetadata();
-            Map<String, String> userData = metadata.getUserMetadata();
-            Content.Builder builder = Content.builder();
-            String type = userData.get("type");
-            if (!type.equals("none")) {
-                builder.withContentType(type);
+            try (S3Object object = s3Client.getObject(request)) {
+                ObjectMetadata metadata = object.getObjectMetadata();
+                Map<String, String> userData = metadata.getUserMetadata();
+                Content.Builder builder = Content.builder();
+                String type = userData.get("type");
+                if (!type.equals("none")) {
+                    builder.withContentType(type);
+                }
+                builder.withContentKey(key);
+                builder.withStream(object.getObjectContent());
```

Comment:
I think this is going to suffer from the same problem of closing the stream in the returned object.